Source/JavaScriptCore/API/JSScript.mm

```diff
@@ -184,7 +184,7 @@ - (void)readCache
 
     SHA1::Digest computedHash;
     SHA1 sha1;
-    sha1.addBytes(fileData, fileDataSize);
```

Comment:
Ideally, we'd be constructing the span as early as possible:
```
auto fileData = mappedFile.span();
// Ensure we at least have a SHA1::Digest to read.
if (fileData.size() < sizeof(SHA1::Digest)) {
    FileSystem::deleteFile(cacheFilename);
    return;
}
SHA1::Digest computedHash;
SHA1 sha1;
sha1.addBytes(fileData.left(fileData.size() - sizeof(SHA1::Digest)));
sha1.computeHash(computedHash);

auto hashSpan = dileData.right(sizeof(SHA1::Digest));
SHA1::Digest fileHash;
memcpy(&fileHash, hashSpan.data(), hashSpan.size());
```

Reply:
Oh, that's indeed much cleaner. I'll switch to that approach.